aten/src/ATen/native/PadNd.cpp

```diff
@@ -163,10 +163,13 @@ Tensor _pad_enum(const Tensor &self, IntArrayRef pad, int64_t mode_int, c10::opt
   if (mode == at::padding_mode::constant) {
     return at::constant_pad_nd(self, pad, value.value_or(0.0));
   }
-  TORCH_CHECK(
-      !value.has_value(), "Padding mode \"",
-      padding_mode_string(mode),
-      "\" doesn't take in value argument");
+  if (value.has_value()) {
+    auto err_msg = c10::str("Padding mode \"", padding_mode_string(mode),
+                            "\" doesn't take in value argument");
+    // Emit error if value != 0, otherwise just warn
+    TORCH_CHECK(*value == 0, err_msg);
+    TORCH_WARN(err_msg);
```

Comment:
Do we really want to deprecate this? Or are we just happy with allowing `0.` here?
Given that people use this already and this is not very harmful. I would vote for not changing the current behavior. At least for now so that we can quickly land this. We can discuss (with Joel) if we want to deprecate that later.

Reply:
I agree that maintaining the current behaviour until we figure out what to do is probably preferable. 

Note that we already have a PR at https://github.com/pytorch/vision/pull/5875 which mitigates the issue by not passing any parameter in specific modes but maintaining BC is worth it for other users.

Reply:
Okay, I've removed the warning. I would argue that a passing parameter that gets ignored is somewhat confusing, but if it's already baked into torchvision then I guess it's too late to fix.